nixos/modules/installer/tools/tools.nix

```
@@ -166,7 +166,7 @@ in
         # Enable CUPS to print documents.
         # services.printing.enable = true;
 
-        # Enable sound.
+        # Enable sound (may be already enabled by a desktop manager like gnome).
```

```suggestion
        # Enable sound and puleaudio. Some desktop manager like gnome use pipewire where both should not be enabled.
```
Before it read like whatever, doesn't hurt but both cannot be enabled.

---

TBF while debugging this I was wondering whether this assertion should not actually read `cfg.enable -> !config.hardware.pulseaudio.enable`, given how subtle the issue was and how much things looked like they were working properly until I tried using a bluetooth headset.

---

Sorry if this looked like I was pointing fingers, I was mostly trying to say that a bluetooth device can be connected (afaiu) only to either pulseaudio or pipewire, and so having the two running at the same time on a single system can lead to hard-to-predict behavior :)

---

You are right about hardware often requiring exclusive access but PipeWire is more than just an audio daemon, and people might still want to use PulseAudio for that functionality for whatever reason (though I think lot of people running nixos-unstable just switched to PipeWire).